packages/dataviews/src/view-table.tsx

```diff
@@ -337,27 +338,11 @@ function TableRow< Item extends AnyItem >( {
 					! isTouchDevice.current &&
 					document.getSelection()?.type !== 'Range'
 				) {
-					if ( ! isSelected ) {
-						onSelectionChange(
-							data.filter( ( _item ) => {
-								const itemId = getItemId?.( _item );
-								return (
-									itemId === id ||
-									selection.includes( itemId )
-								);
-							} )
-						);
-					} else {
-						onSelectionChange(
-							data.filter( ( _item ) => {
-								const itemId = getItemId?.( _item );
-								return (
-									itemId !== id &&
-									selection.includes( itemId )
-								);
-							} )
-						);
-					}
+					onSelectionChange( ( state ) =>
```

Comment:
Can all these calls that use the `updater` be rewritten as?

```js
onSelectionChange(
    selection.includes(id)
        ? selection.filter((itemId) => id !== itemId)
        : [...selection, id]
);
```

I haven't checked them all but it seems that where `onSelectionChange` is available, `selection` also is. If we can, it makes the updater redundant, and the API would simpler without it.

Reply:
> the API would simpler without it

I haven't changed the API at all, these are just React setState calls. Updater functions are nice because, if you're in memoized callbacks, you don't depend on the state, though I've removed all of those. It's also surfaced an issue: if you click on the checkbox, we're calling setState twice. This worked before because it would set state twice with the same value, but with the the updater, the second call is undoing the last call.

Reply:
I see your point, thanks for elaborating 👍

Reply:
Interesting finding regarding the duplicate setState call, I will have a look into that and see if I can fix it.